common/controller_runtime_utils.go

```diff
@@ -61,3 +67,43 @@ func GetWatchNamespace() (string, error) {
 	}
 	return ns, nil
 }
+
+// ReloadRuntimeConfig obtains all values needed by runtime config during Reconcile and writes them to the existing RuntimeConfig provided
+func ReloadRuntimeConfig(ctx context.Context, client client.Client, request ctrl.Request, nodeName string, rc *staticProvisioner.RuntimeConfig) error {
+	// get associated provisioner config
+	cm := &corev1.ConfigMap{}
+	err := client.Get(ctx, types.NamespacedName{Name: ProvisionerConfigMapName, Namespace: request.Namespace}, cm)
+	if err != nil {
+		klog.ErrorS(err, "could not get provisioner configmap", "name", ProvisionerConfigMapName, "namespace", request.Namespace)
+		return err
+	}
+
+	// get current node
+	node := &corev1.Node{}
+	err = client.Get(ctx, types.NamespacedName{Name: nodeName}, node)
+	if err != nil {
+		klog.ErrorS(err, "could not get current Node", "name", nodeName, "namespace", request.Namespace)
+		return err
+	}
+
+	// read provisioner config
+	provisionerConfig := staticProvisioner.ProvisionerConfiguration{}
+	if err := staticProvisioner.ConfigMapDataToVolumeConfig(cm.Data, &provisionerConfig); err != nil {
+		klog.ErrorS(err, "could not load provisioner config from ConfigMap")
+		return err
+	}
+
+	rc.Name = GetProvisionedByValue(*node)
+	rc.Node = node
+	rc.Namespace = request.Namespace
+	rc.DiscoveryMap = provisionerConfig.StorageClassConfig
+	rc.NodeLabelsForPV = provisionerConfig.NodeLabelsForPV
+	rc.SetPVOwnerRef = provisionerConfig.SetPVOwnerRef
+	rc.UseNodeNameOnly = provisionerConfig.UseNodeNameOnly
+	rc.MinResyncPeriod = provisionerConfig.MinResyncPeriod
+	rc.UseAlphaAPI = provisionerConfig.UseAlphaAPI
+	rc.LabelsForPV = provisionerConfig.LabelsForPV
+
```

Comment:
may as well move:

```
r.runtimeConfig.UseJobForCleaning = false
```

Reply:
Done. It was not used in deleter so I did not want touch it. But since it will default to false anyway it should be safe to move it to the helper. Thanks.